src/containers/project-watcher.jsx

```diff
@@ -7,46 +7,60 @@ import {
     getIsShowingWithId
 } from '../reducers/project-state';
 
+/**
+ * Watches for project to finish updating before taking some action.
+ *
+ * To use ProjectWatcher, pass it a callback function using the onDoneUpdating prop.
+ * ProjectWatcher passes a waitForUpdate function to its children, which they can call
+ * to set ProjectWatcher to request that it call the onDoneUpdating callback when
+ * project is no longer updating.
+ */
 class ProjectWatcher extends React.Component {
     constructor (props) {
         super(props);
         bindAll(this, [
-            'setRequesting'
+            'waitForSaving'
         ]);
 
         this.state = {
-            requesting: false
+            waiting: false
         };
     }
     componentDidUpdate (prevProps) {
-        if (this.state.requesting && this.props.isShowingWithId && !prevProps.isShowingWithId) {
-            this.props.onShowingWithId();
-            this.setState({ // eslint-disable-line react/no-did-update-set-state
-                requesting: false
-            });
+        if (this.state.waiting && this.props.isShowingWithId && !prevProps.isShowingWithId) {
+            this.fulfillRequest();
         }
-
     }
-    setRequesting () {
-        this.setState({
-            requesting: true
+    fulfillRequest () {
```

Comment:
Another renaming suggestion (you can ignore). Request is just so overloaded. Would fulfillAction make sense instead as it's fulfilling the user's action to share or see community.

Reply:
How about "fulfill()"?